src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackProcessor.java

```diff
@@ -74,8 +91,18 @@ private void attachCase(String exceptionRecordJurisdiction, String caseRef) {
         return ImmutableList.of(message);
     }
 
-    @NotNull
     private List<String> success() {
         return emptyList();
     }
+
+    @SuppressWarnings({"unchecked", "squid:S1135"})
+    //TODO WIP
+    private Map<String, Object> insertNewScannedDocument(Map<String, Object> exceptionData,
+                                                         Map<String, Object> caseData) {
+        //TODO check SCANNED_DOCUMENTS exists and has a document
+        //TODO check that document Id is unique and not duplicate in caseData
+        List<Object> caseList = (List<Object>) caseData.get(SCANNED_DOCUMENTS);
```

Comment:
why we are using object here ? do we not have a ScannedDocument or something similar ?

Reply:
Nope its all a map of String -> Object
Hence the depth of the validations.

Reply:
The issue is the data section can contain a number of different types of "objects"
there is no union idea in java so the only way to represent this format is to use a json style map.